ui/propertyeditor/propertyeditorfactory.cpp

```diff
@@ -45,6 +45,7 @@ PropertyEditorFactory::PropertyEditorFactory()
     initBuiltInTypes();
 
     addEditor(QVariant::Color, new QStandardItemEditorCreator<PropertyColorEditor>());
+    addEditor(QVariant::ByteArray, new QStandardItemEditorCreator<PropertyTextEditor>());
```

Comment:
I'd expect the QVariant magic to get the correct type.
I tested it with fragmentShader and the changes where applied as expected.